src/runtime/internal/database.server.ts

```diff
@@ -61,12 +61,30 @@ export async function checkAndImportDatabaseIntegrity(event: H3Event, collection
 async function _checkAndImportDatabaseIntegrity(event: H3Event, collection: string, integrityVersion: string, config: RuntimeConfig['content']) {
   const db = loadDatabaseAdapter(config)
 
-  const before = await db.first<{ version: string }>(`select * from ${tables.info} where id = ?`, [`checksum_${collection}`]).catch(() => ({ version: '' }))
+  const before = await db.first<{ version: string, ready: boolean }>(`select * from ${tables.info} where id = ?`, [`checksum_${collection}`]).catch(() => ({ version: '', ready: true }))
+
+  // if another request has already started the initialization of
+  // this version of this collection
+  // wait for it to finish
+  if (before.ready === false && before.version === integrityVersion) {
+    await new Promise((resolve) => {
+      const interval = setInterval(async () => {
+        const { ready } = await db.first<{ ready: boolean }>(`select ready from ${tables.info} where id = ?`, [`checksum_${collection}`]).catch(() => ({ ready: true }))
+        if (ready) {
+          clearInterval(interval)
+          resolve(0)
+        }
+      }, 200)
+    })
+    return true
+  }
```

Comment:
I think it is worth adding a timeout mechanism to this interval to make sure it always resolves.
Think of an edge case in which importing data fails when updating `ready` to `true`

Reply:
totally agree. I added a 5 min timeout if it does not respond. Then I throw.